src/Surfnet/StepupMiddleware/ApiBundle/Identity/Projector/RaCandidateProjector.php

```diff
@@ -157,4 +204,62 @@ protected function applyIdentityForgottenEvent(IdentityForgottenEvent $event)
     {
         $this->raCandidateRepository->removeByIdentityId($event->identityId);
     }
+
+    protected function applySelectRaaOptionChangedEvent(SelectRaaOptionChangedEvent $event)
+    {
+        $authorizedInstitutions = $event->selectRaaOption->getInstitutions($event->institution);
+        $this->updateInstitutionCandidatesFromCollection(new Institution($event->institution->getInstitution()), $authorizedInstitutions);
+    }
+
+    protected function applyInstitutionConfigurationRemovedEvent(InstitutionConfigurationRemovedEvent $event)
+    {
+        $this->raCandidateRepository->removeByRaInstitution(new Institution($event->institution->getInstitution()));
+    }
+
+    /**
+     * @param Institution $institution
+     * @param ConfigurationInstitution[] $authorizedInstitutions
+     * @throws \Doctrine\ORM\NonUniqueResultException
+     */
+    private function updateInstitutionCandidatesFromCollection(Institution $institution, array $authorizedInstitutions)
```

Comment:
In my dev environment, non ra-candidates seem to end up in the `ra-candidates` projection. The search query later correctly filters these options away so the impact on the read side is not evident. I suspect this behaviour is introduced in this PR, but I'm not 100% sure. Can you verify if this is correct?

Reply:
This behavior was already present and this depends on the token verification logic